test/python/utils/mitigation/test_meas.py

```diff
@@ -0,0 +1,707 @@
+            seed_simulator=SEED,
+            seed_transpiler=SEED,
+        ).result()
+
+        # Make a calibration matrix
+        meas_cal = TensoredMeasFitter(cal_results, mit_pattern=mit_pattern)
+        # Calculate the fidelity
+        fidelity = meas_cal.readout_fidelity(0) * meas_cal.readout_fidelity(1)
+
+        results = qiskit.execute(
+            [ghz], backend=backend, shots=self.shots, seed_simulator=SEED, seed_transpiler=SEED
+        ).result()
+
+        # Predicted equally distributed results
+        predicted_results = {"000": 0.5, "111": 0.5}
+
+        meas_filter = meas_cal.filter
+
+        # Calculate the results after mitigation
+        output_results_pseudo_inverse = meas_filter.apply(
+            results, method="pseudo_inverse", meas_layout=meas_layout
+        ).get_counts(0)
+        output_results_least_square = meas_filter.apply(
+            results, method="least_squares", meas_layout=meas_layout
+        ).get_counts(0)
+
+        # Compare with expected fidelity and expected results
+        self.assertAlmostEqual(fidelity, 1.0)
+        self.assertAlmostEqual(
+            output_results_pseudo_inverse["000"] / self.shots, predicted_results["000"], places=1
+        )
+
+        self.assertAlmostEqual(
+            output_results_least_square["000"] / self.shots, predicted_results["000"], places=1
+        )
+
+        self.assertAlmostEqual(
+            output_results_pseudo_inverse["111"] / self.shots, predicted_results["111"], places=1
+        )
+
+        self.assertAlmostEqual(
+            output_results_least_square["111"] / self.shots, predicted_results["111"], places=1
+        )
+
+
+@unittest.skipUnless(HAS_AER, "Qiskit aer is required to run these tests")
+class TestMeasCalNoise(TestMeasCal):
+    """Run measurement mitigation calibration tests with noise."""
+
+    def test_meas_fitter_with_noise(self):
+        """Test the MeasurementFitter with noise."""
+        tests = []
+        runs = 3
+        for run in range(runs):
+            cal_results, state_labels, circuit_results = meas_calibration_circ_execution(
+                1000, SEED + run
+            )
+
+            meas_cal = CompleteMeasFitter(cal_results, state_labels)
+            meas_filter = MeasurementFilter(meas_cal.cal_matrix, state_labels)
+
+            # Calculate the results after mitigation
+            results_pseudo_inverse = meas_filter.apply(circuit_results, method="pseudo_inverse")
+            results_least_square = meas_filter.apply(circuit_results, method="least_squares")
+            tests.append(
+                {
+                    "cal_matrix": convert_ndarray_to_list_in_data(meas_cal.cal_matrix),
+                    "fidelity": meas_cal.readout_fidelity(),
+                    "results": circuit_results,
+                    "results_pseudo_inverse": results_pseudo_inverse,
+                    "results_least_square": results_least_square,
+                }
+            )
```

Comment:
In the Ignis version of this test, these results are loaded from a saved file.  I timed it on my machine, and this whole test takes under a second to run, so I think we're fine to generate the results like this.

Reply:
Yeah it's an annoying pattern in ignis tests where they use json files committed to the repo for testing the fitters where the generation of the data is quite quick. I tried to remove all of that whenever I encountered it in the tests since it was unnecessary complexity (and git repo bloat).